airflow/models/__init__.py

```diff
@@ -3051,8 +3052,8 @@ def __init__(
             )
 
         self.schedule_interval = schedule_interval
-        if isinstance(schedule_interval, Hashable) and schedule_interval in cron_presets:
-            self._schedule_interval = cron_presets.get(schedule_interval)
+        if isinstance(schedule_interval, six.string_types) and schedule_interval in cron_presets:
```

Comment:
This looks like a possible code change. Please double check with the PR that introduced the "Hashable" here for the reasoning

Reply:
It looks like the `Hashable` check was added because `relativedelta` objects aren't hashable in `python-dateutil` before version 2.7. It turns out that the keys of `cron_presets` are always strings, so we can check that `schedule_interval` is a string instead of checking that it's hashable. We could also require `python-dateutil>=2.7` and drop both checks.